src/clients/http_client/__tests__/http_client.test.ts

```diff
@@ -850,7 +876,7 @@ describe('HTTP client', () => {
         body: {errors: 'Error 429'},
         code: 429,
         statusText: 'Error 429',
-        headers: {'X-Text-Header': ['Error 429'], 'Retry-After': ['100']},
+        headers: {'X-Text-Header': 'Error 429', 'Retry-After': '100'},
```

Comment:
(2/3) I'm a bit concerned about removing the `[]` from the comparison to get this to pass ... but this is the format we're expecting/intending to be returned for headers?

Reply:
Now using `canonicalizeHeaders()` when queuing a mocked response.